marketplace/src/benchmarking.rs

```diff
@@ -17,236 +17,166 @@
 #![cfg(feature = "runtime-benchmarks")]
 
 use super::*;
+use crate::Pallet as Marketplace;
 use frame_benchmarking::{account as benchmark_account, benchmarks, impl_benchmark_test_suite};
-use frame_support::{assert_ok, bounded_vec, traits::Currency};
+use frame_support::{assert_ok, traits::Currency};
 use frame_system::RawOrigin;
+use sp_arithmetic::per_things::Permill;
 use sp_runtime::traits::{Bounded, StaticLookup};
 use sp_std::prelude::*;
 use ternoa_common::traits::NFTExt;
 
-use crate::Pallet as Marketplace;
+const PERCENT_50: Permill = Permill::from_parts(500000);
+
+pub fn get_account<T: Config>(name: &'static str) -> T::AccountId {
+	let account: T::AccountId = benchmark_account(name, 0, 0);
+	account
+}
 
-const SERIES_ID: u8 = 20;
+pub fn origin<T: Config>(name: &'static str) -> RawOrigin<T::AccountId> {
+	RawOrigin::Signed(get_account::<T>(name))
+}
 
-pub fn prepare_benchmarks<T: Config>() -> (MarketplaceId, MarketplaceId, NFTId) {
+pub fn prepare_benchmarks<T: Config>() {
```

Comment:
I would suggest you that this function returns the data that is has created.
Something like create a new struct like called  BenchmarksData and let it contain the marketplace id.

I would also suggest you to create an NFT and to return it too via BenchmarksData. In this way you don't need to repeat the same steps for Buy, List and Unlist extrinsics